tests/components/hydrawise/test_config_flow.py

```diff
@@ -0,0 +1,136 @@
+"""Test the Hydrawise config flow."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+import pytest
+from requests.exceptions import ConnectTimeout, HTTPError
+
+from homeassistant import config_entries
+from homeassistant.components.hydrawise.const import DOMAIN
+from homeassistant.const import CONF_API_KEY, CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+@patch("pydrawise.legacy.LegacyHydrawise")
+async def test_form(
+    mock_api: MagicMock, hass: HomeAssistant, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] is None
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"api_key": "abc123"}
+    )
+    mock_api.return_value.status = "All good!"
+    mock_api.return_value.customer_id = 12345
+    await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.CREATE_ENTRY
+    assert result2["title"] == "Hydrawise"
+    assert result2["data"] == {"api_key": "abc123"}
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+@patch("pydrawise.legacy.LegacyHydrawise", side_effect=HTTPError)
+async def test_form_api_error(mock_api: MagicMock, hass: HomeAssistant) -> None:
+    """Test we handle API errors."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"api_key": "abc123"}
+    )
+
+    assert result2["type"] == FlowResultType.ABORT
+    assert result2["reason"] == "cannot_connect"
+
+
+@patch("pydrawise.legacy.LegacyHydrawise", side_effect=ConnectTimeout)
+async def test_form_connect_timeout(mock_api: MagicMock, hass: HomeAssistant) -> None:
```

Comment:
Except import failures these tests should first have the exception, then clear the exception so that in the end these tests ends with `FlowResultType.CREATE_ENTRY`

Reply:
Spot-checking other `test_config_flow.py` files, this does not seem like a common pattern, and I'd argue this makes the tests more difficult to understand and maintain.

Reply:
Done.